lib/graphql/schema/field.rb

```diff
@@ -253,7 +253,7 @@ def initialize(type: nil, name: nil, owner: nil, null: nil, description: :not_gi
         method_name = method || hash_key || name_s
         @dig_keys = dig
         if hash_key
-          @hash_key = hash_key
+          @hash_key = hash_key.to_sym
```

Comment:
If `@hash_key` is a string, hash won't be checked for symbol key otherwise.

Reply:
Honestly, I'd _prefer_ if the only lookup that GraphQL-Ruby did was with the user-provided `hash_key`. But, early versions of GraphQL-Ruby added the string-based lookup, and ever since, I have used `@hash_key_str` to maintain compatibility. But I think it would be better if it only used the configured `hash_key:`, without any conversions. 

For this `.to_sym`, is this to retain compatibility with an old GraphQL-Ruby behavior? If not, I'd rather not include it in this fix. It might even break if anyone is using numbers as hash keys.

Reply:
I see. I wasn't sure what this is about. I should have asked. In general I agree in being as explicit as possible so will remove that.

Reply:
```suggestion
          @hash_key = hash_key
```

Reply:
Sorry for being late. Removed the according tests.